suite-native/graph/src/components/Graph.tsx

```diff
@@ -1,27 +1,53 @@
-import React from 'react';
+import React, { useMemo } from 'react';
 import { LineGraph, GraphPoint } from 'react-native-graph';
 
 import { prepareNativeStyle, useNativeStyles } from '@trezor/styles';
-import { Box } from '@suite-native/atoms';
+import { Box, Text } from '@suite-native/atoms';
+import { getExtremaFromGraphPoints, sumLineGraphPoints } from '@suite-common/wallet-graph';
+
+import { AxisLabel } from './AxisLabel';
 
 type GraphProps = {
     points: GraphPoint[];
 };
 
 const graphWrapperStyle = prepareNativeStyle(_ => ({
-    width: '100%',
-    height: 150,
-    marginVertical: 40,
+    height: 250,
+    justifyContent: 'center',
+    alignItems: 'center',
 }));
 
 const graphStyle = prepareNativeStyle(_ => ({
     alignSelf: 'center',
-    aspectRatio: 21 / 9,
+    aspectRatio: 1.4,
+    width: '100%',
 }));
 
 export const Graph = ({ points = [] }: GraphProps) => {
     const { applyStyle } = useNativeStyles();
 
+    const nonZeroSumOfGraphPoints = useMemo(() => sumLineGraphPoints(points) > 0, [points]);
```

Comment:
We should avoid premature perf optimization/memoization. If you want to really perf optimization first thing you must do is to measure and benchmark to have baseline. Another good rule is not to do any advanced optimization/memoization unless you are experiencing issue.

For example `JSON.stringify` is probably more expensive operation that `sumLineGraphPoints` itself (JSON operations are very slow in general).

About the other solution I don't know without measurements, but from brief look to implementation `sumLineGraphPoints` I think impact will be very small, maybe even negative because there is just some simple `map` and `reduce` to sum numbers so it will be very fast unless there will be bigger tens of thousands of points.

In case you really want to do perf optimization, you should start with optimization of alogorithm itself and fallback to memoization as last choice. 

In this example of `sumLineGraphPoints` you can probably omit first `map` and do that directly in `reduce` which will save you one whole cycle of going though all of points. Another optimization could be using ts-belt `reduce` function because [it's about 60% faster](https://mobily.github.io/ts-belt/benchmarks/v3.12.0/macbook-pro-2021#reduce) than native `.reduce` implementation. 

If after all of these optimalizations, function still will be slow you can think of momization or other techniques.

btw. maybe we can include this ^^^ some text like this into code style guide to hooks section about `useMemo` etc.

Reply:
absolutely, we should have something on optimization. I mentioned JSON, because I think with `points` as dep it will be re-rendered much more often. Graph is one of those things which could definitely be optimized from the start IMO, since it's working with a lot of data by design. At least with things like less iterations, which you and I mentioned :)